src/main/java/frc/robot/classes/spikemotor/SpikeMotor.java

```diff
@@ -0,0 +1,252 @@
+package frc.robot.classes.spikemotor;
+
+import java.util.ArrayList;
+
+/**
+* An abstracted motor interface designed for easy replacement.
+*/
+public abstract class SpikeMotor {
+    private boolean isInitialized = false;
+    private ArrayList<SpikeMotor> followers;
+    private SpikeMotor follower = null;
+
+    /**
+    * Initializes this motor with the given device number.
+    *
+    * @param deviceNumber    an int giving the device number
+    * @return                nothing
+    * @throws                IllegalStateException if already initialized
+    * @see                   #initImpl(int deviceNumber)
+    */
+    public final void init(int deviceNumber) {
+        if (isInitialized)
+            throw new IllegalStateException(
+                "Motor cannot be initialized twice"
+            );
+        followers = new ArrayList<SpikeMotor>();
+        initImpl(deviceNumber);
+        isInitialized = true;
+    }
+
+    /**
+    * Follows another motor.
+    *
+    * @param motor  the SpikeMotor to follow
+    * @return       nothing
+    * @throws       IllegalStateException if either motor is not initialized
+    */
+    public final void follow(SpikeMotor otherMotor) {
+        if (!isInitialized)
+            throw new IllegalStateException(
+                "Motor must be initialized first"
+            );
+        if (!otherMotor.isInitialized)
+            throw new IllegalStateException(
+                "Cannot follow a motor that has not been initialized"
+            );
+        if (otherMotor.follower != null) {
```

Comment:
Let's make this a method call instead of acting on the member of another class directly. 
`otherMotor.AddFollower(this);`

Reply:
One reason for doing this is that we may want to do something or trigger something when a follower is added. Let's say the other motor does not have the capability to have followers

Reply:
Can do!